components/sup/doc/http_gateway_services_schema.json

```diff
@@ -0,0 +1,521 @@
+        "required": [
+          "butterfly_data_path",
+          "census_data_path",
+          "composites_path",
+          "data_path",
+          "member_id_file",
+          "proc_lock_file",
+          "services_data_path",
+          "specs_path",
+          "sup_root"
+        ],
+        "properties": {
+          "butterfly_data_path": {
+            "type": "string",
+            "description": "The path for butterfly.dat"
+          },
+          "census_data_path": {
+            "type": "string",
+            "description": "The path for census.dat"
+          },
+          "composites_path": {
+            "type": "string",
+            "description": "The path for composites"
+          },
+          "data_path": {
+            "type": "string",
+            "description": "The path for data"
+          },
+          "member_id_file": {
+            "type": "string",
+            "description": "The path to the MEMBER_ID file"
+          },
+          "proc_lock_file": {
+            "type": "string",
+            "description": "The path to the LOCK file"
+          },
+          "services_data_path": {
+            "type": "string",
+            "description": "The path for services.dat"
+          },
+          "specs_path": {
+            "type": "string",
+            "description": "The path to the supervisor spec files"
+          },
+          "sup_root": {
+            "type": "string",
+            "description": "The supervisor's root directory"
+          }
+        }
+      },
+      "needs_reconfiguration": {
+        "type": "boolean",
+        "description": "Does this service need to be reconfigured"
+      },
+      "needs_reload": {
+        "type": "boolean",
+        "description": "Does this service need to be reloaded"
+      },
+      "pkg": {
+        "type": "object",
+        "description": "The habitat package that this service was spawned from",
+        "required": [
+          "deps",
+          "env",
+          "exports",
+          "exposes",
+          "ident",
+          "name",
+          "origin",
+          "path",
+          "release",
+          "svc_config_path",
+          "svc_data_path",
+          "svc_files_path",
+          "svc_group",
+          "svc_path",
+          "svc_pid_file",
+          "svc_run",
+          "svc_static_path",
+          "svc_user",
+          "svc_var_path",
+          "version"
+        ],
+        "properties": {
+          "deps": {
+            "type": "array",
+            "description": "This package's dependencies",
+            "items": {
+              "description": "Package identifier for a dependency",
+              "$ref": "#/definitions/package_identifier"
+            }
+          },
+          "env": {
+            "type": "object",
+            "description": "The environment for this package",
+            "additionalProperties": {
+              "type": ["null", "boolean", "string", "integer", "array", "object"]
+            }
```

Comment:
I don't think this is correct... `env` is an object of environment variables, so it should have String keys and values (might need to check how it works with values that are numbers, booleans, etc... do they actually get encoded as Strings?)

Reply:
You're right.  I have no idea what I was thinking.